pep-0634.rst

```diff
@@ -0,0 +1,697 @@
+variable in the nearest function scope unless there's an applicable
+``nonlocal`` or ``global`` statement.)
+
+In a given pattern, a given name may be bound only once.  This
+disallows e.g. ``case x, x: ...`` but allows ``case [x] | x: ...``.
+
+.. _wildcard_pattern:
+
+Wildcard Pattern
+~~~~~~~~~~~~~~~~
+
+Syntax::
+
+    wildcard_pattern: "_"
+
+A wildcard pattern always succeeds.  It binds no name.
+
+.. _constant_value_pattern:
+
+Constant Value Patterns
+~~~~~~~~~~~~~~~~~~~~~~~
+
+TODO: Rename to Value Patterns?  (But ``value[s]_pattern`` is already
+a grammatical rule.)
+
+Syntax::
+
+  constant_pattern: attr
+  attr: name_or_attr '.' NAME
+  name_or_attr: attr | NAME
+
+The dotted name in the pattern is looked up using the standard Python
+name resolution rules.  However, when the same constant pattern occurs
+multiple times in the same ``match`` statement, the interpreter may cache
+the first value found and reuse it, rather than repeat the same
+lookup.  (To clarify, this cache is strictly tied to a given execution
+of a given ``match`` statement.)
+
+The pattern succeeds if the value found thus compares equal to the
+subject value (using the ``==`` operator).
+
+
+Group Patterns
+~~~~~~~~~~~~~~
+
+Syntax:
+
+  group_pattern: '(' pattern ')'
+
+(For the syntax of ``pattern``, see Patterns above.)
+
+A parenthesized pattern has no additional syntax.  It allows users to
+add parentheses around patterns to emphasize the intended grouping.
+
+
+.. _sequence_pattern:
+
+Sequence Patterns
+~~~~~~~~~~~~~~~~~
+
+
+Syntax::
+
+  sequence_pattern:
+    | '[' [values_pattern] ']'
+    | value_pattern ',' [values_pattern]
+    | '(' ')'
+
+(For ``values_pattern`` and ``value_pattern``, see Patterns above.)
+
+(Note that a single parenthesized pattern without a trailing comma is
+a group pattern, not a sequence pattern.)
+
+A sequence pattern may (directly) contain at most one subpattern of
+the form ``'*' NAME``; all other subpatterns must be walrus patterns,
+OR patterns or closed patterns.
+
+A sequence pattern fails if the subject value is not an instance of
+``collections.abc.Sequence``.  It also fails if the subject value is
+an instance of ``str``, ``bytes`` or ``bytearray``.
+
+If the one of the subpatterns has the form ``'*' NAME``, this is
+called a variable-length sequence pattern.  A variable-length sequence
+pattern fails if the length of the subject sequence is less than the
+number of subpatterns not of that form.
+
+If no such subpattern is present, the sequence pattern is considered
+fixed-length.  A fixed-length sequence pattern fails if the length of
+the subject sequence is not equal to the number of subpatterns.
+
+The length of the subject sequence is obtained using the builtin
+``len()`` function (i.e., via the ``__len__`` protocol).  However, the
+interpreter may cache this value in a similar manner as described for
+constant value patterns.
+
+TODO: Do we have left-to-right semantics here?
```

Comment:
Yes. Actually, *everything* is left-to-right except walrus patterns (which in my mind is a decent argument for switching to `as` syntax).

I'll remove this line.